urlchecker/core/urlmarker.py

```diff
@@ -50,7 +50,7 @@
         ")",
         "|[a-z0-9.\\-]+[.](?:%s)/)" % domain_extensions,
         "(?:",
-        "[^\\s()<>\[\\]]+|\\([^\\s()]*?\\([^\\s()]+\\)[^\\s()]*?\\)",
+        r"[^\\s()<>\[\\]]+|\\([^\\s()]*?\\([^\\s()]+\\)[^\\s()]*?\\)",
```

Comment:
Everything other than this looks good to me. I will check this in a couple of hours from now and see if I can come up with a quick fix.

Reply:
Thank you! This was my effort to fix the warning - turning into a raw string (what I did above) helps sometimes. But the tests are still failing - it's not even detecting URLs on many cases, so we have a larger issue on our hands. 

I appreciate your help @SuperKogito !

Reply:
So the good news, you had the right fix to the warning and we have some good tests. 
The bad news, we got a lot of deprecation warnings and some tough URLs to check e.g. https://codepen.io/rootwork/  cannot be checked from my machine but the link is there.
 
Also check this, this is the same test, https://groups.drupal.org/node/278968 causes the fails at the first one and then works in the second. Zero code changed in between just timing I guess.
![ucheck1](https://github.com/urlstechie/urlchecker-python/assets/15731839/e2ec84f5-a8e6-4f31-b170-ced8f16d1c12)
![ucheck2](https://github.com/urlstechie/urlchecker-python/assets/15731839/c84acb2b-02a2-469c-82c8-a42a3088de7f)


My suggestions are the following: 
- Using your fix but all over the `URL_REGEX` in https://github.com/urlstechie/urlchecker-python/blob/7dbd7ac171cf85788728b4cf5576c191f13c8399/urlchecker/core/urlmarker.py#L45

```python3

URL_REGEX = r"".join(
    (
        r"(?i)\b(",
        r"(?:",
        r"https?:(?:/{1,3}|[a-z0-9%]",
        r")",
        r"|[a-z0-9.\-]+[.](?:%s)/)" % domain_extensions,
        r"(?:",
        r"[^\s()<>\[\]]+|\([^\s()]*?\([^\s()]+\)[^\s()]*?\)",
        r"|\([^\\s]+?\)",
        r")",
        r"+",
        r"(?:",
        r"\([^\s()]*?\([^\s()]+\)[^\s()]*?\)",
        r"|\([^\\s]+?\)",
        r"|[^\s`!()\[\];:'\".,<>?«»“”‘’]",
        r")",
        r"|",
        r"(?:",
        r"(?<!@)[a-z0-9]",
        r"+(?:[.\-][a-z0-9]+)*[.]",
        r"(?:%s)\b/?(?!@)" % domain_extensions,
        r"))",
    )
)
```

- Comment the difficult links causing an issue -for now- until we figure a better way to check them (This is not a REGEX issue imo ... I can only point the finger to the Driver atm :/ ). In my case commenting out ("https://groups.drupal.org/node/298298" and  "https://codepen.io/rootwork/") under https://github.com/urlstechie/urlchecker-python/blob/7dbd7ac171cf85788728b4cf5576c191f13c8399/tests/test_core_check.py#L41
made the test pass for Python 3.9 and 3.12.

- A key point is to extend the test.yml to test for different versions of Python (The warning for the escape char -among others- only shows up on Python 3.12 and not on 3.9). In this regard, we need to decide which Python versions do we want to support and which ones we want to gradually drop. I think our test.yml 'test section' should look a bit more like one https://github.com/librosa/librosa/blob/main/.github/workflows/ci.yml
This way we cover more versions. 

*I did not make any direct changes to the branch since I don't have a better solution and your input on these matters is very important.